mongo/integration/sdam_error_handling_test.go

```diff
@@ -78,16 +79,59 @@ func TestSDAMErrorHandling(t *testing.T) {
 				mt.ResetClient(clientOpts)
 				clearPoolChan()
 
-				// The saslContinue blocks for 150ms so run the InsertOne with a 100ms context to cause a network
-				// timeout during auth and assert that the pool was cleared.
+				// Use a context with a 100ms timeout so that the saslContinue delay of 150ms causes
+				// an operation-scoped context timeout (i.e. a timeout not caused by a client timeout
+				// like connectTimeoutMS or socketTimeoutMS).
 				timeoutCtx, cancel := context.WithTimeout(mtest.Background, 100*time.Millisecond)
 				defer cancel()
 				_, err := mt.Coll.InsertOne(timeoutCtx, bson.D{{"test", 1}})
 				assert.NotNil(mt, err, "expected InsertOne error, got nil")
 				assert.True(mt, mongo.IsTimeout(err), "expected timeout error, got %v", err)
 				assert.True(mt, mongo.IsNetworkError(err), "expected network error, got %v", err)
+				assert.False(mt, isPoolCleared(), "expected pool not to be cleared but was cleared")
```

Comment:
This assertion errored in [this task](https://evergreen.mongodb.com/lobster/evergreen/test/mongo_go_driver_tests_42_plus_zlib_zstd_support__version~latest_os_ssl_40~ubuntu1804_64_go_1_16_test_replicaset_auth_nossl_patch_4adb6d73ca0fe53c99e11765005f37033acd6471_60d56ac30ae60671e32a3517_21_06_25_05_33_57/0/ac6e3fcaffebeef411d9d724df4dcfe6#bookmarks=0%2C19006&l=1&shareLine=11759).

Similar to my comment below, my guess is that there is a race in the order here:

```
mt.ResetClient(clientOpts)
clearPoolChan()
```

Though that begs the question of why this was not failing before.

Reply:
Will look into any race conditions with `poolMonitor`.